### PR TITLE
feat(batch): ec2 launch template support

### PR DESCRIPTION
**Pull request by andrestone**
**!** Integration tests are failing on master, waiting for this [fix](https://github.com/aws/aws-cdk/pull/6549) to be merged.

### Related issues:
- https://github.com/aws/aws-cdk/pull/4732#issuecomment-593167267

### Usage
Simply define your Launch Template:
```
    const myLaunchTemplate = new ec2.CfnLaunchTemplate(this, 'LaunchTemplate', {
      launchTemplateName: 'extra-storage-template',
      launchTemplateData: {
        blockDeviceMappings: [
          {
            deviceName: '/dev/xvdcz',
            ebs: {
              encrypted: true,
              volumeSize: 100,
              volumeType: 'gp2'
            }
          }
        ]
      }
    });
```
and use it:

```
    const myComputeEnv = new batch.ComputeEnvironment(this, 'ComputeEnv', {
      computeResources: {
        launchTemplate: {
          launchTemplateName: myLaunchTemplate.launchTemplateName as string, //or simply use an existing template name
        },
        vpc,
      },
      computeEnvironmentName: 'MyStorageCapableComputeEnvironment',
    });
```

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-batch/lib/compute-environment.ts [hunk 1]
```diff
@@ -46,6 +46,22 @@ export enum AllocationStrategy {
...
+ */
+export interface LaunchTemplateSpecification {
+  /**
+   * The Launch template name
+   */
+  readonly launchTemplateName: string;
+  /**
+   * The launch template version to be used (optional).
+   *
+   * @default the default version of the launch template
```
Comment (iliapolo) [inline packages/@aws-cdk/aws-batch/lib/compute-environment.ts]: use `-` :)

File: packages/@aws-cdk/aws-batch/lib/compute-environment.ts [hunk 2]
```diff
@@ -57,6 +73,16 @@ export interface ComputeResources {
    */
   readonly instanceRole?: iam.IRole;
 
+  /**
+   * An optional launch template to associate with your compute resources. To use a launch template, you must
+   * specify either the launch template ID or launch template name in the request, but not both.
```
Comment (iliapolo) [inline packages/@aws-cdk/aws-batch/lib/compute-environment.ts]: This is a bit confusing, the interface doesn't include an ID option. Also, `request` is not the correct wording here. I suggest you just drop this sentence all together. Its clear enough without it.

File: packages/@aws-cdk/aws-batch/lib/compute-environment.ts [hunk 3]
```diff
@@ -57,6 +73,16 @@ export interface ComputeResources {
...
+   * For more information, see Launch Template Support.
```
Comment (iliapolo) [inline packages/@aws-cdk/aws-batch/lib/compute-environment.ts]: `see Launch Template Support`? Did you mean to link to the README content you added?

**Comment by iliapolo** [comment]
@andrestone I love it, so clean!

File: packages/@aws-cdk/aws-batch/lib/compute-environment.ts (hunk 3 above)
Comment (andrestone) [reply packages/@aws-cdk/aws-batch/lib/compute-environment.ts]: My bad, some heavy copy pasting on my side. I'll fix.

**Comment by andrestone** [comment]
I guess it's mergeable now? 
Thanks!

**Comment by iliapolo** [comment]
> I guess it's mergeable now?
> Thanks!

This wasn't merged yet: https://github.com/aws/aws-cdk/pull/6549

Also the PR description thing

**Comment by andrestone** [comment]
Wouldn't the squash commit be ok as well?

**Comment by iliapolo** [comment]
> Wouldn't the squash commit be ok as well?

Whoops sorry, didn't notice it. Yeap all good. You just need to re-run CodeBuild project by pushing a dummy commit.

**Comment by andrestone** [comment]
Maaaaaa bad. Gotta remake expected.json

**Comment by iliapolo** [comment]
Are you running ‘yarn integ’ to produce the expected template?

**Comment by andrestone** [comment]
yep

**Comment by andrestone** [comment]
Build is failing on master after this commit: https://github.com/aws/aws-cdk/commit/25952a9c04769654c09bac83421683220f32acb7

**Pull request merged**